test/no_only.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+var tap = require('tap');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var concat = require('concat-stream');
+
+var stripFullStack = require('./common').stripFullStack;
+
+var tapeBin = path.join(process.cwd(), 'bin/tape');
+
+tap.test(
+    'Should throw error when --no-only is passed via cli and there is a .only test',
+    { todo: process.versions.node.match(/0\.10\.\d+/) ? 'Fails on node 0.10' : false },
+    function (tt) {
+        tt.plan(2);
+
+        var testStderr = function (rows) {
+            tt.ok((/Error: there should be no only test\n/).test(stripFullStack(rows.toString('utf8')).join('\n')));
+        };
+
+        var ps = spawn(tapeBin, ['**/*js', '--no-only' ], { cwd: path.join(__dirname, 'no_only') });
+        ps.stderr.pipe(concat(testStderr));
+        ps.on('exit', function (code) {
+            tt.equal(code, 1);
+        });
+    }
+);
+
+tap.test(
+    'Should throw error when NODE_TAPE_NO_ONLY_TEST is passed via envs and there is an .only test',
+    { todo: process.versions.node.match(/0\.10\.\d+/) ? 'Fails on node 0.10' : false },
+    function (tt) {
+        tt.plan(2);
+
+        var testStderr = function (rows) {
+            tt.ok((/Error: there should be no only test\n/).test(stripFullStack(rows.toString('utf8')).join('\n')));
+        };
+
+        process.env.NODE_TAPE_NO_ONLY_TEST = true;
+        var ps = spawn(tapeBin, ['**/*.js' ], { cwd: path.join(__dirname, 'no_only') });
```

Comment:
rather than mutating the env, let's set it in the spawn call.

Reply:
done